src/plugins/kernel_install/utils.sh

```diff
@@ -86,19 +95,29 @@ function list_installed_kernels()
 
   while read -r kernel; do
     if [[ -f "$prefix/boot/vmlinuz-$kernel" && ! "$kernel" =~ .*\.old$ ]]; then
+      if sudo grep -q "$kernel" "$managed_kernels_path"; then
+        managed_kernels+=("$kernel")
+      fi
       available_kernels+=("$kernel")
     fi
   done <<< "$output"
 
-  echo
+  if [[ -z "$all" ]]; then
+    if [[ "${#managed_kernels[@]}" == 0 ]]; then
+      echo 'No kernels installed are managed by kw.' \
+        'Pass --all|-a to see all installed kernels'
+      return 0
+    else
+      available_kernels=("${managed_kernels[@]}")
+    fi
+  fi
 
   if [[ "$single_line" != 1 ]]; then
     printf '%s\n' "${available_kernels[@]}"
   else
     echo -n "${available_kernels[0]}"
-    available_kernels=("${available_kernels[@]:1}")
-    printf ',%s' "${available_kernels[@]}"
-    echo ""
+    printf ',%s' "${available_kernels[@]:1}"
```

Comment:
When I tested your patch, I deploy one single test in the remote, and when I used:
```
kw d -s
```
I got
```
5.13.0-joao-kernel+,
```
It looks like that your patch is adding one extra `,` at the end. Can we remove that?